src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_t1/key_vault_authentication.py

```diff
@@ -59,7 +59,7 @@ def __init__(self, authorization_callback):
     # for backwards compatibility we need to support callbacks which don't accept the scheme
     def _auth_callback_compat(self, server, resource, scope, scheme):
         return self._user_callback(server, resource, scope) \
-            if len(inspect.getargspec(self._user_callback).args) == 3 \
+            if len(inspect.signature(self._user_callback).parameters) == 3 \
```

Comment:
This is not equivalent to the original code if `self._user_callback` contains `*args` and `**kwargs`, since [`signature`](https://docs.python.org/3.10/library/inspect.html#inspect.signature)'s result includes `*args` (`kind`: `VAR_POSITIONAL`) and `**kwargs` (`kind`: `VAR_KEYWORD`) in `parameters`, but [`getargspec`](https://docs.python.org/3.10/library/inspect.html#inspect.getargspec)'s result has dedicated `varargs`, `keywords` elements for them:

```py
import inspect

def testfunc(a=1, b=2, *args, **kwargs):
    pass

print(len(inspect.getfullargspec(testfunc).args))
print(inspect.getfullargspec(testfunc).args)
print(len(inspect.signature(testfunc).parameters))
print(inspect.signature(testfunc).parameters)
```

Output:

```
2
['a', 'b']
4
OrderedDict([('a', <Parameter "a=1">), ('b', <Parameter "b=2">), ('args', <Parameter "*args">), ('kwargs', <Parameter "**kwargs">)])
```